subpages/lake_modelling.py

```diff
@@ -1,12 +1,209 @@
+
+    lime_dose, pct_ca, od_fac, K_L = getLimProd()
+
+    n_months = getDuration()
+
+    lake_model, x_title, y_title = model(
+        lime_dose,
+        pct_ca,
+        od_fac,
+        C_bott0,
+        C_lake0,
+        area,
+        mean_depth,
+        mean_flow,
+        C_in0,
+        K_L,
+        n_months,
+        f,
+    )
+    # st.write(lake_model)
+
+    plotResult(lake_model, x_title, y_title)
+
+    return None
+
+
+def getLakeParams():
+    st.markdown("### Lake characteristics")
+    col1, col2 = st.columns(2)
+
+    C_lake0 = col1.number_input(
+        "Initial lake Ca concentration (mg-Ca/l)", min_value=0, value=1
     )
+    C_in0 = col1.number_input(
+        "Lake inflow Ca concentration (mg-Ca/l)", min_value=0, value=1
+    )
+    C_bott0 = col1.number_input(
+        "Lake bottom concentration (mg-Ca/l)", min_value=0, value=0
+    )
+    area = col2.number_input("Lake area (km2)", min_value=0.0, value=1.14)
+    mean_depth = col2.number_input("Mean lake depth (m)", min_value=0.0, value=5.6)
+    mean_flow = col2.number_input("Mean flow rate (m3/s)", min_value=0.0, value=0.2)
+
+    return (C_lake0, C_in0, C_bott0, area, mean_depth, mean_flow)
+
+
+def getLimProd():
+    # define data for liming products (later separate database?)
+    name = "Name"
+    dose = "Dose"
+    content = "Content"
+    overdosing = "Overdosing"
+    dissolution = "Dissolution"
+    data = {
+        name: ["Product 1", "Product 2", "Custom"],
+        dose: [10, 35, 0],
+        content: [38.5, 15, 0],
+        overdosing: [1.5, 2, 0],
+        dissolution: [1, 1, 0],
+    }
+    df = pd.DataFrame(data=data)
+
+    st.markdown("### Liming product")
+    limProd = st.selectbox("Choose liming product", (df))
+    prod = df.loc[df[name] == limProd]
+
+    if limProd == "Custom":
+        return getCustomLimParam()
+    else:
+        st.write(
+            "You chose ",
+            limProd,
+            " with lime dose = ",
+            prod[dose].values[0],
+            "(mg-lime/l), lime Ca content = ",
+            prod[content].values[0],
+            ", %, Overdosing factor = ",
+            prod[overdosing].values[0],
+            " and rate of dissolution of lake bottom lime = ",
+            prod[dissolution].values[0],
+            ".",
+        )
+        return (
+            prod[dose].values[0],
+            prod[content].values[0],
+            prod[overdosing].values[0],
+            prod[dissolution].values[0],
+        )
+
+
+def getCustomLimParam():
+    st.markdown("#### Custom Liming parameters")
+    col1, col2 = st.columns(2)
+    lime_dose = col1.number_input('Liming "dose" (mg-lime/l)', min_value=0, value=10)
+    pct_ca = col1.number_input("Lime Ca content by mass (%)", min_value=0.0, value=38.5)
+    od_fac = col2.number_input("Overdosing factor (-)", min_value=0.0, value=1.5)
+    K_L = col2.number_input(
+        "Lime dissoltuion rate on the bottom of the lake (month^-1)",
+        min_value=1,
```

Comment:
Minor comment: the `min_value` for $K_L$ should be zero, not one.

Reply:
Fixed